Task/86 - Incorporate Icon Component in Flagship App

matt-guest-wilcore opened the pull request:
## Description of Change
<!-- Please include a description of the change and context. What would a code reviewer, or a future dev, 
need to know about this PR in order to understand why this PR was created? This could include dependencies 
introduced, changes in behavior, pointers to more detailed documentation. The description should be more 
than a link to an issue.  -->

Updated every instance of `<VAIcon />` in the app to the new `<Icon />` component from the mobile-component-library dependancy. Doing so also required finding and updating the names of the assets to their new equivalent (ex: RightArrowInCircle is now ArrowCircleRight). 

I also had to update a few tests after adding the Icon component to some screens. For the components `BaseListItem` and `VASelector` I had to incorporate the new vads colors in order to get icons to show properly (Some didn't mesh well with the previous theme system).

I made an effort to check every screen in the app (in light and dark mode) to make sure the new icons are showing properly. I'd recommend also going through the app yourself in case I missed anything. I'm going to try to get the UX designers to have a look at this and get their input as well.

## Screenshots/Video
<!-- Add screenshots or video as needed. Before/after if changes are to be compared by reviewers.
Before/after: <img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" />
Toggle: <details><summary></summary><img src="" width="49%" />&nbsp;&nbsp;<img src="" width="49%" /></details>
-->

## Testing
<!-- What testing was done to verify the changes (local/unit)? What testing remains? Note edge cases, or special
situations that could not be tested during development. -->

- [x] Tested on iOS <!-- simulator is fine -->
- [ ] Tested on Android <!-- simulator is fine -->

## Reviewer Validations
<!-- What should reviewers look for? Copy/paste Acceptance Criteria from ticket -->

## PR Checklist
<!-- Engineer: make sure all these items are checked off before requesting a review -->
  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

kimmccaskill-oddball commented:
![Screenshot 2025-02-11 at 3 06 08 PM](https://github.com/user-attachments/assets/5ccfb51f-860b-481c-b3f2-c9f7ac701aec)
![Screenshot 2025-02-11 at 3 06 48 PM](https://github.com/user-attachments/assets/3efd6723-2124-4891-906f-8db8ca413fcb)
While everything seems to be working, I just want to make sure the differences are intentional. Thanks!

matt-guest-wilcore commented:
Yes, some icons will look different. The new Icon library sometimes doesn't have the same icons as the previous so I had to follow the Figma for guidance. Here's an [example](https://www.figma.com/design/Zzt8z60hCtdEzXx2GFWghH/%F0%9F%93%90-Component-Library---Design-System---VA-Mobile?node-id=235-674&t=cAE23rAtl3d0xNuN-0) showing the icons used for links in the app using the new library. Because the VA.gov link opens up an external web page the icon was changed to reflect that.

I'm thinking I should get UX to take a look at this just to verify everything is updated as it should. Thanks for bringing this up.

kimmccaskill-oddball reviewed (dismissed):
Gonna go ahead and approve the code pending design approval ✅

nhuckleberry6 commented:
@matt-guest-wilcore @kimmccaskill-oddball thanks for finding those and calling them out! Good catch on removing the icon next to "Only use messages for non-urgent needs" since it's not an external link. The other couple items I found are [here in Figma](https://www.figma.com/design/P1MgY68Y7Jju5PCfaJHQX4/Misc.-MFS-Stuff?node-id=803-25617&t=K2DvYD2ZDCzD8ZHe-1). The checkmark is more important to fix than the slight sizing and alignment issue of the attach icon. Otherwise, nothing else was jumping out at me.
<img width="1140" alt="Screenshot 2025-02-13 at 2 16 24 PM" src="https://github.com/user-attachments/assets/9ea19840-8184-4e69-923b-7e6ce57c5558" />

kimmccaskill-oddball commented:
While Natasha mentioned that the sizing was more important, I still think we shouldn't introduce the alignment issues here  Here's the screenshot now that the icon is bigger
develop(before)
![Screenshot 2025-02-17 at 10 14 55 AM](https://github.com/user-attachments/assets/35fce71c-4c42-4d13-b224-9e36fa910c1a)

current branch(after)
![Screenshot 2025-02-17 at 10 12 53 AM](https://github.com/user-attachments/assets/f5ef271b-d493-46d5-8965-7bedf5d6744a)

matt-guest-wilcore commented:
I discussed with Natasha and updated the code to center align the icon and the text. I also adjusted the size of the paperclip icon to make it a bit smaller. I believe it looks a lot better now.

kimmccaskill-oddball reviewed (approved):
Looks good!
